### PR TITLE
build: rm luarocks tar file behind successful installation

### PR DESCRIPTION
**Pull request by kwanhur**
### What this PR does / why we need it:

remove `luarocks` downloaded tar file

1. successfully install it, no need to reserve it.
2. install script remove the untar files (line#45)[https://github.com/apache/apisix/blob/master/utils/linux-install-luarocks.sh#L45], so remove the tar file simultaneously.

### Pre-submission checklist:

* [x] Did you explain what problem does this PR solve? Or what new features have been added?
* [ ] Have you added corresponding test cases?
* [ ] Have you modified the corresponding document?
* [ ] Is this PR backward compatible? **If it is not backward compatible, please discuss on the [mailing list](https://github.com/apache/apisix/tree/master#community) first**

### CODE REVIEW

File: utils/linux-install-luarocks.sh [hunk 1]
```diff
@@ -42,7 +42,7 @@ fi
 make build > build.log 2>&1 || (cat build.log && exit 1)
 sudo make install > build.log 2>&1 || (cat build.log && exit 1)
 cd .. || exit
-rm -rf luarocks-"$LUAROCKS_VER"
+rm -rf luarocks-"$LUAROCKS_VER" luarocks-"$LUAROCKS_VER".tar.gz
```
Comment (spacewander) [inline utils/linux-install-luarocks.sh:45]: Could we remove it just after untar?

Comment (kwanhur) [reply utils/linux-install-luarocks.sh:45]: > Could we remove it just after untar?

on this way, if `configure` failed, it need to download archive file `v3.8.0.tar.gz` repeatedly.

Comment (spacewander) [reply utils/linux-install-luarocks.sh:45]: OK. So that we can also handle https://github.com/apache/apisix/issues/6307 in this PR.

Comment (kwanhur) [reply utils/linux-install-luarocks.sh:45]: no, #6307 occurs at `configure` failure, so download twice.

Comment (spacewander) [reply utils/linux-install-luarocks.sh:45]: Yes. But they change the same thing twice. So why not put together?

Comment (kwanhur) [reply utils/linux-install-luarocks.sh:45]: ok, remove it behind untar was better, I'll update this logic.

File: utils/linux-install-luarocks.sh [hunk 2]
```diff
@@ -25,6 +25,7 @@ fi
 LUAROCKS_VER=3.8.0
 wget https://github.com/luarocks/luarocks/archive/v"$LUAROCKS_VER".tar.gz
 tar -xf v"$LUAROCKS_VER".tar.gz
+rm -f luarocks-"$LUAROCKS_VER".tar.gz
```
Comment (leslie-tsang) [inline utils/linux-install-luarocks.sh:28]:
```suggestion
rm -f v"$LUAROCKS_VER".tar.gz
```

**Review by leslie-tsang — CHANGES_REQUESTED** [review]

**Review by leslie-tsang — APPROVED** [review]

**Review by spacewander — APPROVED** [review]

**Pull request merged**